src/test/java/com/kodilla/ecommercee/ItemTestSuite.java

```diff
@@ -0,0 +1,168 @@
+package com.kodilla.ecommercee;
+
+import com.kodilla.ecommercee.domain.*;
+import com.kodilla.ecommercee.repository.*;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest()
+public class ItemTestSuite {
+
+    @Autowired
+    private ItemRepository itemRepository;
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Test
+    public void testCreateAndReadItem_And_RelatedEntities() {
+        //Given
+        Group group = new Group("meble");
+
+        groupRepository.save(group);
+
+        Product product = new Product("stolik", "mały stolik", BigDecimal.valueOf(299L), group);
+        Cart cart = new Cart(new BigDecimal(1500));
+
+        Item item1 = new Item(4,product,cart);
+        Item item2 = new Item(1,product,cart);
+
+        List<Item> newItemsList = new ArrayList<>();
+        newItemsList.add(item1);
+        newItemsList.add(item2);
+        product.setItems(newItemsList);
+
+        cart.getItems().add(item1);
+        cart.getItems().add(item2);
+
+        //When
+        cartRepository.save(cart);
+        itemRepository.save(item1);
+        itemRepository.save(item2);
```

Comment:
```suggestion
        cartRepository.save(cart);
```

Myślę, że można usunąć zapisywanie itemów, bo zapisują się razem z cartem dzięki kaskadzie. U mnie test przeszedł mimo wszystko - zobacz czy u Ciebie też działa.

Reply:
Ok, dzięki zerknę :)